### PR TITLE
Attach signature in Custom Section without mutating module.

### PR DESCRIPTION
**Pull request by PiotrSikora**
Signed-off-by: Piotr Sikora <piotrsikora@google.com>

### CODE REVIEW

**Comment by PiotrSikora** [comment]
@jedisct1 PTAL

File: src/lib.rs [hunk 1]
```diff
@@ -20,11 +19,7 @@ pub fn sign(
     symbol_name: &str,
 ) -> Result<Vec<u8>, WError> {
     let signature_alg = key_pair.sk.to_alg()?;
-    let module: Module = parity_wasm::deserialize_buffer(module_bytes)?;
-    let module =
-        wasm_signature::attach_signature(module, &signature_alg, ad, &key_pair, symbol_name)?;
-    let signed_module_bytes = parity_wasm::serialize(module)?;
-    Ok(signed_module_bytes)
+    wasm_signature::attach_signature(&module_bytes, &signature_alg, ad, &key_pair, symbol_name)
```
Comment (PiotrSikora) [inline src/lib.rs:22]: I moved serialization/deserialization bits to `wasm_signature` as well for the old signatures to keep things consistent. Let me know if you prefer me to revert this part.

**Comment by jedisct1** [comment]
Looking good!

Thank you!

File: src/lib.rs (hunk 1 above)
Comment (jedisct1) [reply src/lib.rs:22]: No problem! Consistency is great.

**Pull request merged**